node/middlewares/internals/saveInternalCategoryRoute.ts

```diff
@@ -41,48 +42,68 @@ const getInternal = (
   type: PAGE_TYPES[type],
 })
 
-const identifyCategory = async (
+const saveCategoryInternal = async (
+  identifiedCategory: IdentifiedCategory,
+  clients: Clients
+) => {
+  const { rewriterGraphql, apps } = clients
+  const { type, params, id, map } = identifiedCategory
+  const path = await getPath(PAGE_TYPES[type], params, apps)
+  const internal: InternalInput = getInternal(path, type, id, map)
+
+  await rewriterGraphql.saveInternal(internal)
+}
+
+const saveCategoryTree = async (
   category: Category,
-  catalogGraphQL: CatalogGraphQL
-): Promise<IndentifiedCategory> => {
+  clients: Clients
+): Promise<IdentifiedCategory> => {
+  const { catalogGraphQL } = clients
   const { parentCategoryId, name } = category
   if (!parentCategoryId) {
-    return {
+    const identifiedCategory = {
+      id: category.id,
       map: 'c',
       params: {
         department: slugify(name!),
       },
-      type: 'DEPARTMENT',
+      type: 'DEPARTMENT' as CategoryTypes,
     }
+    await saveCategoryInternal(identifiedCategory, clients)
+    return identifiedCategory
   }
 
   const parentCategory = await catalogGraphQL
     .category(parentCategoryId)
     .then(prop('category'))
-  const { type, params, map } = await identifyCategory(
-    parentCategory,
-    catalogGraphQL
-  )
+  const { type, params, map } = await saveCategoryTree(parentCategory, clients)
   if (type === 'DEPARTMENT') {
-    return {
+    const identifiedCategory = {
+      id: parentCategory.id,
       map: `${map},c`,
       params: {
         ...params,
         category: slugify(name!),
       },
-      type: 'CATEGORY',
+      type: 'CATEGORY' as CategoryTypes,
     }
+    await saveCategoryInternal(identifiedCategory, clients)
+    return identifiedCategory
   } else if (type === 'CATEGORY') {
-    return {
+    const identifiedCategory = {
+      id: parentCategory.id,
       map: `${map},c`,
       params: {
         ...params,
         subcategory: slugify(name!),
       },
-      type: 'SUBCATEGORY',
+      type: 'SUBCATEGORY' as CategoryTypes,
     }
+    await saveCategoryInternal(identifiedCategory, clients)
+    return identifiedCategory
   } else {
-    return {
+    const identifiedCategory = {
```

Comment:
What's the type that's not category or department?

Reply:
It's subcategory. Do you think it is better to  be explicit?

Reply:
I think so, the else makes we think that there are more